zerver/tests/test_user_groups.py

```diff
@@ -868,7 +888,9 @@ def check_create_user_group(acting_user: str, error_msg: Optional[str] = None) -
             acting_user=None,
         )
         check_create_user_group("cordelia", "Insufficient permission")
+        expected_group_cnt += 2
         check_create_user_group("shiva")
+        expected_group_cnt -= 1
```

Comment:
I am not sure about the readability of this. But can we atleast move the ` expected_group_cnt -= 1` line after we delete the group to make it clear why we are reducing the count. And also might be worth to add a comment before at least one ` expected_group_cnt += 2` line that it is due to the new single user group being created.